spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/TraceIdLoggingWebMvcInterceptor.java

```diff
@@ -38,12 +38,41 @@
  */
 public class TraceIdLoggingWebMvcInterceptor extends HandlerInterceptorAdapter {
 
-	private static final String X_CLOUD_TRACE = "x-cloud-trace-context";
+	public static final String X_CLOUD_TRACE_HEADER = "x-cloud-trace-context";
 
-	private static final String X_B3_TRACE = "X-B3-TraceId";
+	public static final String X_B3_TRACE_HEADER = "X-B3-TraceId";
 
-	private static final List<String> TRACE_HEADERS = ImmutableList.of(X_CLOUD_TRACE,
-			X_B3_TRACE);
+	private static final List<String> DEFAULT_HEADERS = ImmutableList.of(X_CLOUD_TRACE_HEADER,
+			X_B3_TRACE_HEADER);
+
+	private final List<String> traceHeaders;
+
+	private final boolean allowOnlyOneHeader;
+
+	/**
+	 * Constructor that specifies the headers to check in what order, and if to only
+	 * expect a single header to be in the request.
+	 * @param headers the headers to check in order.
+	 * @param allowOnlyOneHeader true to allow multiple headers with trace IDs, false to
+	 * throw an exception if more than one header is found to have trace ID.
+	 */
+	public TraceIdLoggingWebMvcInterceptor(List<String> headers,
```

Comment:
I'm not sure how autowire-friendly this constructor is.
It seems like it would be more appropriate in Spring to just turn headers into a bean and replace `DEFAULT_HEADERS` with a bean that is created by auto-configuration.

Reply:
the `getLoggingWebMvcInterceptor` function in `StackdriverLoggingAutoConfiguration` class just returns an instance of this right? Couldn't we check if a setting like `spring.cloud.gcp.logging.onlyoneheader` and use that in the constructor?

Reply:
I think the list of headers itself can be a bean, and in `StackdriverLoggingAutoConfiguration` we would create it with the two default headers, if one is not defined by the user. In that case, we probably don't even need the `onlyoneheader` option because, the user can just define a bean with one header to be used in the `LoggingWebMvcInterceptor`.

Reply:
" the user can just define a bean with one header to be used "
But wouldn't this be a loss of functionality? You mentioned originally throwing an exception, which implies the user should know about an unexpected situation of having both headers when he only expected one. If the user only ever provides one header name, then we wouldn't be able to check for this.

In fact I kind of question the need for the ability to specify headers in spring settings files at all. Isn't the selling point of this project that GCP services specifically use the `x-cloud-content-trace` and `x-b3-trace` headers?

Reply:
One might argue that "the selling point" of Spring is openness to customization and configuration. :) Auto-configuration in Spring Boot can provide the two default headers, but still keep the class open for configuration. If more than one search header has a value, we need to decide what to do -- fail or use first value silently. I don't have a strong opinion on that.
Let's discuss offline.

Reply:
PTAL, I changed it up as discussed. I will make the springboot config changes in the following PR ,so right now even though all the classes and such are added it's still using the GCP header like before.